### PR TITLE
fix(kselect): clear filter on blur [KHCP-13444]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-13444

* clear KSelect filter on blur
* create new item on Enter key press when filter returns no items
  * mirror this change in KMultiselect for UX consistency across the two components

### CODE REVIEW

File: src/components/KMultiselect/KMultiselect.vue [hunk 1]
```diff
@@ -815,9 +817,15 @@ const handleItemSelect = (item: MultiselectItem, isNew?: boolean) => {
...
+const onInputEnter = (): void => {
+  if (!filteredItems.value.length && props.enableItemCreation) {
+    handleAddItem()
+  }
+}
+
 // add an item with `enter`
 const handleAddItem = (): void => {
-  if (!props.enableItemCreation || !filterString.value || !uniqueFilterStr.value) {
+  if (!props.enableItemCreation || !filterString.value || !uniqueFilterStr.value || !props.itemCreationValidator(filterString.value)) {
```
Comment (Justineo) [inline src/components/KMultiselect/KMultiselect.vue:828]: non blocking: It seems that we are missing `itemCreationValidator` in `<KSelect>`, shall we align this?

**Review by Justineo — APPROVED** [review]
LGTM! Thank you!

File: src/components/KMultiselect/KMultiselect.vue (hunk 1 above)
Comment (portikM) [reply src/components/KMultiselect/KMultiselect.vue:828]: Already have a PR for this: https://github.com/Kong/kongponents/pull/2485

**Review by Justineo — APPROVED** [review]
LGTM 🚀

**Pull request merged**